File: app/src/main/java/com/zskjprojectj/andouclient/activity/mall/PinTuanFragment.java
package com.zskjprojectj.andouclient.activity.mall;

import android.os.Bundle;
import android.view.View;

import androidx.annotation.Nullable;
import androidx.recyclerview.widget.RecyclerView;

import com.chad.library.adapter.base.BaseQuickAdapter;
import com.zhuosongkj.android.library.app.BaseFragment;
import com.zskjprojectj.andouclient.R;
import com.zskjprojectj.andouclient.adapter.mall.PinTuanGoodsAdapter;
import com.zskjprojectj.andouclient.model.PinTuan;

import butterknife.BindView;

public class PinTuanFragment extends BaseFragment {

    @BindView(R.id.recyclerView)
    RecyclerView recyclerView;

    PinTuanGoodsAdapter adapter = new PinTuanGoodsAdapter();

    public PinTuanFragment(PinTuan.PinTuanType pinTuanType) {
        this.pinTuanType = pinTuanType;
    }

    private PinTuan.PinTuanType pinTuanType;

    @Override
    public void onActivityCreated(@Nullable Bundle savedInstanceState) {
        super.onActivityCreated(savedInstanceState);
        adapter.bindToRecyclerView(recyclerView);
        adapter.setNewData(pinTuanType.goods);

        adapter.setOnItemChildClickListener(new BaseQuickAdapter.OnItemChildClickListener() {
            @Override
            public void onItemChildClick(BaseQuickAdapter adapter, View view, int position) {
                MallGoodsDetailsActivity.start("23");
            }
        });

    }

    @Override
    protected int getContentView() {
        return R.layout.framgent_pin_tuan;
    }
}

File: app/src/main/java/com/zskjprojectj/andouclient/activity/mall/PinTuanFragment.kt
package com.zskjprojectj.andouclient.activity.mall

import android.os.Bundle
import com.chad.library.adapter.base.BaseQuickAdapter
import com.zhuosongkj.android.library.app.BaseFragment
import com.zhuosongkj.android.library.util.PageLoadUtil
import com.zskjprojectj.andouclient.R
import com.zskjprojectj.andouclient.adapter.mall.PinTuanGoodsAdapter
import com.zskjprojectj.andouclient.http.ApiUtils
import com.zskjprojectj.andouclient.model.PinTuan.PinTuanType
import kotlinx.android.synthetic.main.framgent_pin_tuan.*

class PinTuanFragment(private val pinTuanType: PinTuanType) : BaseFragment() {
    var adapter = PinTuanGoodsAdapter()
    override fun onActivityCreated(savedInstanceState: Bundle?) {
        super.onActivityCreated(savedInstanceState)
        adapter.onItemChildClickListener =
                BaseQuickAdapter.OnItemChildClickListener { _, _, position: Int ->
                    MallGoodsDetailsActivity.start(adapter.getItem(position)?.goods_id)
                }
        val pagetUtil = PageLoadUtil.get(mActivity, recyclerView, adapter, refreshLayout)
        pagetUtil.load {
            ApiUtils.getApiService().pinTuanGoods(pinTuanType.id, pagetUtil.page)
        }
    }

    override fun getContentView() = R.layout.framgent_pin_tuan
}